whiterabbit/src/main/java/org/ohdsi/whiteRabbit/scan/UniformSamplingReservoir.java

```diff
@@ -0,0 +1,217 @@
+package org.ohdsi.whiteRabbit.scan;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * Uniform sampling reservoir for streaming. This should capture the input distribution in order
+ * to compute quartiles, using so-called Algorithm R.
+ *
+ * <p>The maximum size of the reservoir can be increased to get more accurate quartile estimations.
+ * As long as the number of samples is lower than the maximum size of the reservoir, the quartiles
+ * are computed exactly. Otherwise they are an estimate. The minimum, maximum and average are always
+ * calculated exactly.
+ */
+public class UniformSamplingReservoir {
+    private double[] samples;
+    private int maxSize;
+    private long populationCount;
+    private double populationSum;

```

Comment:
If populations get really big, it may make sense to use a BigDecimal for this sum to avoid loss of precision due to overflow.